### PR TITLE
gh-124969: Fix locale.nl_langinfo(locale.ALT_DIGITS)

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
* Issue: gh-124969

----
📚 Documentation preview 📚: https://cpython-previews--124974.org.readthedocs.build/

### CODE REVIEW

File: Modules/_localemodule.c [hunk 1]
```diff
@@ -666,9 +666,34 @@ _locale_nl_langinfo_impl(PyObject *module, int item)
...
+                            Py_CLEAR(pyresult);
+                            break;
+                        }
+                        PyTuple_SET_ITEM(pyresult, i, unicode);
+                        result += strlen(result) + 1;
+                    }
+                }
+            }
+#endif
+            else {
```
Comment (methane) [inline Modules/_localemodule.c]: This `else` should be in `#ifdef ALT_DIGITS` block.

**Review by methane — APPROVED** [review]

**Comment by serhiy-storchaka** [comment]
Thank you for your review @methane. I wonder what happened with ALT_DIGITS on macOS whether it only for Japanese (which have lesser need of this info for `strftime()`/`strptime()`) or for all locales. In any case this is no our fault.

**Comment by encukou** [comment]
This commit broke the iOS buildbot: https://buildbot.python.org/#/builders/1380/builds/1535/steps/10/logs/stdio

**Comment by serhiy-storchaka** [comment]
Thanks. Should be fixed by #125177.

**Pull request merged**